iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -1056,7 +1065,7 @@ public void OnConnectionDisconnected()
             if (Logging.IsEnabled)
                 Logging.Enter(this, nameof(OnConnectionDisconnected));
 
-            _amqpAuthenticationRefresher?.StopLoop();
+            _ = _amqpAuthenticationRefresher?.StopLoopAsync().ConfigureAwait(false);
```

Comment:
This, and below event handler:

This event handler cannot await on async code. This handler is generally invoked as a result of the "Close" call. Since the close call with stop and await the token refresher cancellation first, this is merely added as an extra level of check.